bmlt-meeting-list.php

```diff
@@ -2130,15 +2213,35 @@ function filter_plugin_actions($links, $file) {
 		* Retrieves the plugin options from the database.
 		* @return array
 		*/
-		function getMLOptions() {
+		function getMLOptions($current_setting) {
+			if ($current_setting<1 and is_admin( )) {
+		        $current_setting = 1;
+		    }
+		    if ($current_setting != 1) {
+		        $this->optionsName = $this->genoptionName($current_setting);
+		    } else {
+		        $this->optionsName = Bread::OPTIONS_NAME;
+		    }
 			//Don't forget to set up the default options
 			if (!$theOptions = get_option($this->optionsName)) {
+				if ($current_setting != 1) {
+			        unset($this->allSettings[$current_setting]);
+			        update_option(Bread::SETTINGS,$this->allSettings);
+			        die('Undefined setting: '.$current_setting);
+			    }
 				$import_file = plugin_dir_path(__FILE__) . "includes/three_column_settings.json";
 				$encode_options = file_get_contents($import_file);
 				$theOptions = json_decode($encode_options, true);
 				update_option( $this->optionsName, $theOptions );
 			}
 			$this->options = $theOptions;
+			$this->loaded_setting = $current_setting;
+		}
+        /**
+         * @param current_setting
+         */private function genOptionName($current_setting)
```

Comment:
can we call this `getOptionName`?  also the casing does not match throughout.

Reply:
Ah, Java programmer here.  Not a getter because it takes an argument.   I don't know PHP coding conventions, as maybe is obvious to you.  In any case, I can change this and also the casing when I change isset.

Reply:
ok renamed to `generate`, no need to abbreviate, we aren't worried about saving a few bytes here.